crates/binstalk-downloader/src/remote/resolver.rs

```diff
@@ -23,42 +26,75 @@ impl Resolve for TrustDnsResolver {
     }
 }
 
-fn new_resolver() -> Result<TokioAsyncResolver, Box<dyn std::error::Error + Send + Sync>> {
+#[instrument(level = "trace")]
+fn new_resolver() -> Result<TokioAsyncResolver, BoxError> {
     #[cfg(unix)]
     {
+        info!("Using system DNS resolver configuration");
         Ok(TokioAsyncResolver::tokio_from_system_conf()?)
     }
     #[cfg(windows)]
     {
+        info!("Using custom DNS resolver configuration");
         let mut config = ResolverConfig::new();
         let opts = ResolverOpts::default();
 
-        let current_interface = default_net::get_default_interface()?;
-        ipconfig::get_adapters()?
-            .iter()
-            .filter_map(|adapter| {
-                if adapter.adapter_name() == current_interface.name {
-                    Some(adapter.dns_servers())
-                } else {
-                    None
-                }
-            })
-            .flatten()
-            .for_each(|addr| {
-                let socket_addr = SocketAddr::new(*addr, 53);
-                for protocol in [Protocol::Udp, Protocol::Tcp] {
-                    config.add_name_server(NameServerConfig {
-                        socket_addr,
-                        protocol,
-                        tls_dns_name: None,
-                        trust_negative_responses: false,
-                        #[cfg(feature = "rustls")]
-                        tls_config: None,
-                        bind_addr: None,
-                    })
-                }
-            });
+        get_dns_servers()?.into_iter().for_each(|addr| {
+            trace!("Adding DNS server: {}", addr);
+            let socket_addr = SocketAddr::new(addr, 53);
+            for protocol in [Protocol::Udp, Protocol::Tcp] {
+                config.add_name_server(NameServerConfig {
+                    socket_addr,
+                    protocol,
+                    tls_dns_name: None,
+                    trust_negative_responses: false,
+                    #[cfg(feature = "rustls")]
+                    tls_config: None,
+                    bind_addr: None,
+                })
+            }
+        });
 
+        debug!("Resolver configuration complete");
         Ok(TokioAsyncResolver::tokio(config, opts))
     }
 }
+
+#[cfg(windows)]
+#[instrument(level = "trace")]
+fn get_dns_servers() -> Result<Vec<std::net::IpAddr>, BoxError> {
```

Comment:
I would like it to `ip_config::Adapter` to avoid the heap allocation:

```suggestion
fn get_dns_servers() -> Result<ip_config::Adapter, BoxError> {
```

Reply:
Makes sense, will do!